extensions/wikia/CrosslinkTag/CrosslinkTagHelper.class.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/**
+ * Class CrosslinkTagHelper
+ */
+class CrosslinkTagHelper extends WikiaModel {
+
+	const CACHE_TTL = 3600;
+	const FANDOM_API_URL = 'http://fandom.wikia.com/wp-json/wp/v2/';
+	const VALID_HOST = 'fandom.wikia.com';
+
+	protected $endpoints = [
+		'articles' => 'posts',
+		'videos' => 'video',
+	];
+
+	/**
+	 * Show the crosslink unit only on Article pages and Main pages
+	 * @return boolean
+	 */
+	public function canShowUnit() {
+		if ( !$this->app->checkSkin( 'oasis' ) ) {
+			return false;
+		}
+
+		$title = $this->wg->Title;
+		if ( $title instanceof Title && $title->getNamespace() == NS_MAIN ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get data for the article by slug
+	 * @param string $slug - alphanumeric id
+	 * @param string $pageType - page type
+	 * @return array|false
+	 */
+	public function getArticleDataBySlug( $slug, $pageType = 'articles' ) {
+		$cacheKey = $this->getMemcKey( $slug );
+		$data = $this->wg->Memc->get( $cacheKey );
+		if ( !is_array( $data ) ) {
+			$params = [
+				'_embed' => 1,
+				'slug' => $slug,
+			];
+
+			if ( array_key_exists( $pageType, $this->endpoints ) ) {
+				$endpoint = $this->endpoints[$pageType];
+			} else {
+				$endpoint = $this->endpoints['articles'];
+			}
+
+			$apiUrl = self::FANDOM_API_URL . $endpoint.'?' . http_build_query( $params );
+			$method = 'GET';
+			$options = [ 'noProxy' => true ];
+
+			$response = Http::request( $method, $apiUrl, $options );
+			if ( $response === false ) {
+				return false;
+			}
+
+			$data = [];
+			$response = json_decode( $response, true );
+			if ( !empty( $response[0]['id'] ) ) {
+				$result = array_pop( $response );
+
+				$data = [
+					'title' => html_entity_decode( $result['title']['rendered'] ),
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "rendered" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)